src/Nethermind/Nethermind.Synchronization/FastSync/TreeSync.cs

```diff
@@ -133,7 +138,8 @@ public SyncResponseHandlingResult HandleResponse(StateSyncBatch? batch)
 
             try
             {
-                lock (_handleWatch)
+                _pendingRequestsLock.EnterReadLock();
```

Comment:
This is confusing. It enters `ReadLock` but changes things. I expect the assumption is that those changes are thread-safe. 
1. Looks ok:
- `_pendingRequests` as it is `ConcurrentDictionary`.
- `_lastReview` 
2. Looks like a potential access issue:
- `_pendingItems.RecalculatePriorities` 
- `_hintsToResetRoot`
- `_branchProgress` in `AddNodeToPending`
- `_branchProgress` in `SaveNode`
- `_branchProgress` in `HandleTrieNode`
- `_data`
- `_handleWatch` (not Interlocked)
3. Other questions:
- Do we need to call `StoreProgressInDb` every time?

Reply:
Seems like I missed a comment here. I guess it was skipped on the various rebase I did?. Its a readwritelock, but not really a read-write operation. It just need to have exclusive access during the write (as in cleaning up the whole syns state), while allowing concurrent read (during evrything else). Specifically, it protects the five field below it. I guess a better name would be `_syncStateLock`.

Reply:
ok but now everything under `Read` lock is accessed concurrently and I mentioned in point `2` all the fields that might break on that access.

Reply:
Done

Reply:
Done